ntoskrnl/mm/section.c

```diff
@@ -4279,131 +4279,98 @@ NtQuerySection(
     {
-        switch (SectionInformationClass)
+        case SectionBasicInformation:
         {
-            case SectionBasicInformation:
-            {
-                PSECTION_BASIC_INFORMATION Sbi = (PSECTION_BASIC_INFORMATION)SectionInformation;
+            SECTION_BASIC_INFORMATION Sbi;
 
-                _SEH2_TRY
-                {
-                    Sbi->Attributes = 0;
-                    if (Section->u.Flags.Image)
-                        Sbi->Attributes |= SEC_IMAGE;
-                    if (Section->u.Flags.File)
-                        Sbi->Attributes |= SEC_FILE;
-                    if (Section->u.Flags.NoChange)
-                        Sbi->Attributes |= SEC_NO_CHANGE;
-
-                    if (Section->u.Flags.Image)
-                    {
-                        Sbi->BaseAddress = 0;
-                        Sbi->Size.QuadPart = 0;
-                    }
-                    else
-                    {
-                        Sbi->BaseAddress = (PVOID)((PMM_SECTION_SEGMENT)Section->Segment)->Image.VirtualAddress;
-                        Sbi->Size.QuadPart = ((PMM_SECTION_SEGMENT)Section->Segment)->Length.QuadPart;
-                    }
+            Sbi.Size = Section->SizeOfSection;
+            Sbi.BaseAddress = (PVOID)Section->Address.StartingVpn;
 
-                    if (ResultLength != NULL)
-                    {
-                        *ResultLength = sizeof(SECTION_BASIC_INFORMATION);
-                    }
-                    Status = STATUS_SUCCESS;
-                }
-                _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
-                {
-                    Status = _SEH2_GetExceptionCode();
-                }
-                _SEH2_END;
+            Sbi.Attributes = 0;
+            if (Section->u.Flags.Commit)
+                Sbi.Attributes |= SEC_COMMIT;
+            if (Section->u.Flags.Reserve)
+                Sbi.Attributes |= SEC_RESERVE;
+            if (Section->u.Flags.File)
+                Sbi.Attributes |= SEC_FILE;
+            if (Section->u.Flags.Image)
+                Sbi.Attributes |= SEC_IMAGE;
 
-                break;
+            /* FIXME : Complete/test the list of flags passed back from NtCreateSection */
+
+            if (Section->u.Flags.Image)
+            {
+                Sbi.BaseAddress = 0;
+                Sbi.Size.QuadPart = 0;
+            }
+            else if (MiIsRosSectionObject(Section))
+            {
+                Sbi.BaseAddress = (PVOID)((PMM_SECTION_SEGMENT)Section->Segment)->Image.VirtualAddress;
+                Sbi.Size.QuadPart = ((PMM_SECTION_SEGMENT)Section->Segment)->Length.QuadPart;
+            }
+            else
+            {
+                DPRINT1("Unimplemented code path!");
             }
 
-            case SectionImageInformation:
+            _SEH2_TRY
+            {
+                *((SECTION_BASIC_INFORMATION*)SectionInformation) = Sbi;
+                if (ResultLength)
+                    *ResultLength = sizeof(Sbi);
```

Comment:
I'd prefer we keep those braces (here and in the Sii case below). I _think_ it's part of our official style guide now, but who knows.

Reply:
Alright.